### PR TITLE
[onnx] support for lowering mod op from onnx to torch

### PR DESCRIPTION
**Pull request by pavs315**
nod-ai/Shark-Turbine#267

### CODE REVIEW

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp [hunk 1]
```diff
@@ -977,4 +977,16 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
...
+                  Torch::ValueTensorType resultType;
+                  Value lhs, rhs;
+                  if (binder.tensorOperands(lhs, rhs) ||
+                      binder.tensorResultType(resultType)) {
+                    return failure();
+                  }
+                  rewriter.replaceOpWithNewOp<Torch::AtenFmodTensorOp>(
+                      binder.op, resultType, lhs, rhs);
+                  return success();
+                });
```
Comment (vivekkhandelwal1) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp:991]: Hi @pavs315, you haven't handled this attribute https://onnx.ai/onnx/operators/onnx__Mod.html#attributes in the lowering.

**Review by vivekkhandelwal1 — CHANGES_REQUESTED** [review]

**Comment by vivekkhandelwal1** [comment]
Hi @pavs315, this op has been on your plate for more than a month. Can you please try to wrap this up?

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp (hunk 1 above)
Comment (pavs315) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp]: @vivekkhandelwal1 Hi, I have added the fmod attribute in the latest commit, please review.

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp [hunk 2]
```diff
@@ -977,4 +977,22 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
...
+                  if (binder.tensorOperands(lhs, rhs) ||
+                      binder.tensorResultType(resultType) || 
+                      binder.s64IntegerAttr(fmod, "fmod", 0)) {
+                    return failure();
+                  }
+                  Value constFmod = rewriter.create<Torch::ConstantIntOp>(
+                		    binder.getLoc(), rewriter.getType<Torch::IntType>(),
+                		    rewriter.getIntegerAttr(rewriter.getIntegerType(64), fmod));
+                  rewriter.replaceOpWithNewOp<Torch::AtenFmodTensorOp>(
+                      binder.op, resultType, lhs, rhs, constFmod);
```
Comment (vivekkhandelwal1) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp:995]: Hi @pavs315, this piece of code is wrong. The `AtenFmodTensorOp` takes only two operands. Refer: https://github.com/llvm/torch-mlir/blob/89e02c195b910621246c55003fca86558162c6da/include/torch-mlir/Dialect/Torch/IR/GeneratedTorchOps.td#L11593-L11615
Here, you need to put a condition based on the value of `fmod` attribute. If it's 0 then you should create `AtenRemainderTensorOp` and if it's 1 then you should create `AtenFmodTensorOp`.

File: test/Conversion/TorchOnnxToTorch/simple_ops_g_to_p.mlir [hunk 3]
```diff
@@ -582,6 +582,15 @@ func.func @test_globalaveragepool_precomputed(%arg0: !torch.vtensor<[1,1,3,3],f3
 
 // -----
 
+// CHECK-LABEL: func.func @test_mod_int64_fmod
+func.func @test_mod_int64_fmod(%arg0: !torch.vtensor<[6],si64>, %arg1: !torch.vtensor<[6],si64>) -> !torch.vtensor<[6],si64> attributes {torch.onnx_meta.ir_version = 7 : si64, torch.onnx_meta.opset_version = 13 : si64, torch.onnx_meta.producer_name = "backend-test", torch.onnx_meta.producer_version = ""} {
+    // CHECK: torch.aten.fmod.Tensor %arg0, %arg1 : !torch.onnx.fmod<[6],si64>, !torch.vtensor<[6],si64> -> !torch.vtensor<[6],si64>
+    %0 = torch.operator "onnx.Mod"(%arg0, %arg1) {torch.onnx.fmod = 1 : si64} : (!torch.vtensor<[6],si64>, !torch.vtensor<[6],si64>) -> !torch.vtensor<[6],si64>
```
Comment (vivekkhandelwal1) [inline test/Conversion/TorchOnnxToTorch/simple_ops_g_to_p.mlir:588]: Add one more test for the case `fmod = 0`.

**Review by vivekkhandelwal1 — CHANGES_REQUESTED** [review]

**Comment by vivekkhandelwal1** [comment]
@pavs315, please rebase your branch, it has some conflicts.

**Comment by pavs315** [comment]
@vivekkhandelwal1 Hi, my apologies, I can't access my build, so I was not able to access the generated tests. I am attempting to build again, which is taking quite a bit of time.

**Comment by vivekkhandelwal1** [comment]
> @vivekkhandelwal1 Hi, my apologies, I can't access my build, so I was not able to access the generated tests. I am attempting to build again, which is taking quite a bit of time.

Hi @pavs315, have you figured it out or need help?

**Review by vivekkhandelwal1 — APPROVED** [review]

**Pull request merged**